core/src/text/textBuffer.cpp

```diff
@@ -91,15 +96,28 @@ bool TextBuffer::addLabel(const TextStyle::Parameters& _params, Label::Transform
     uint32_t stroke = (_params.strokeColor & 0x00ffffff) + (strokeWidth << 24);
 
     for (auto& q : quads) {
+        // the atlas resolution at which the quad uvs have been generated onto
+        glm::vec2 atlasResolution(q.atlasWidth, q.atlasHeight);
+        glm::vec2 uvScaleFactor(1.f, 1.f);
+
+        if (atlasResolution != m_atlasRes) {
+            uvScaleFactor = atlasResolution / m_atlasRes;
+        }
```

Comment:
The uvScaleFactor of previously added vertices to this mesh will be invalid when the texture size changes during the build of a tile. I think a better approach would be use the pixel offset in the texture as uv which stays constant after resize, so that one only has to divide uv/textureSize in the shader.